pkg/network/usm/ebpf_gotls.go

```
@@ -191,60 +223,28 @@ func newGoTLSProgram(c *config.Config) *GoTLSProgram {
 
 	p.binAnalysisMetric = libtelemetry.NewCounter("gotls.analysis_time", libtelemetry.OptStatsd)
 
-	return p
+	return p, nil
 }
 
-func (p *GoTLSProgram) Name() string {
+func (p *goTLSProgram) Name() string {
 	return "go-tls"
 }
 
-func (p *GoTLSProgram) IsBuildModeSupported(mode buildMode) bool {
-	return mode == CORE || mode == RuntimeCompiled
-}
-
-func (p *GoTLSProgram) ConfigureManager(m *errtelemetry.Manager) {
+func (p *goTLSProgram) ConfigureOptions(m *manager.Manager, options *manager.Options) {
 	p.manager = m
-	p.manager.Maps = append(p.manager.Maps, []*manager.Map{
-		{Name: offsetsDataMap},
-		{Name: goTLSReadArgsMap},
-		{Name: goTLSWriteArgsMap},
-	}...)
-	// Hooks will be added in runtime for each binary
-}
-
-func (p *GoTLSProgram) ConfigureOptions(options *manager.Options) {
 	options.MapSpecEditors[connectionTupleByGoTLSMap] = manager.MapSpecEditor{
 		Type:       ebpf.Hash,
 		MaxEntries: p.cfg.MaxTrackedConnections,
 		EditorFlag: manager.EditMaxEntries,
 	}
 }
 
-func (*GoTLSProgram) GetAllUndefinedProbes() []manager.ProbeIdentificationPair {
-	probeList := make([]manager.ProbeIdentificationPair, 0)
-	for _, probeInfo := range functionToProbes {
-		if probeInfo.functionInfo != "" {
-			probeList = append(probeList, manager.ProbeIdentificationPair{
-				EBPFFuncName: probeInfo.functionInfo,
-			})
-		}
-
-		if probeInfo.returnInfo != "" {
-			probeList = append(probeList, manager.ProbeIdentificationPair{
-				EBPFFuncName: probeInfo.returnInfo,
-			})
-		}
-	}
-
-	return probeList
-}
-
-func (p *GoTLSProgram) Start() {
+func (p *goTLSProgram) PreStart(m *manager.Manager) error {
 	var err error
-	p.offsetsDataMap, _, err = p.manager.GetMap(offsetsDataMap)
+	p.offsetsDataMap, _, err = m.GetMap(offsetsDataMap)
 	if err != nil {
 		log.Errorf("could not get offsets_data map: %s", err)
-		return
+		return err
```

return the error instead of logging and returning an error